File: Abstr.java
interface Student{
   public void prepare();
}
class Sem implements Student{
        public void prepare(){
       System.out.println("Students prepare for sem");
   }
}
class Mid implements Student{
        public void prepare(){
       System.out.println("Students won't prepare seriously for mid");
   }
}
class Abstr{
public static void main(String[] args){
   Mid k=new Mid();
   k.prepare();
   Sem a=new Sem();
   a.prepare();
   }
}

File: Abstr.cpp
#include<iostream>
using namespace std;

class Student{
    public:
    virtual void prepare()=0;
};
class Sem:public Student{
    public:
    void prepare(){
        cout<<"Students prepare for sem"<<endl;
    }
};
class mid : public Student{
    public:
    void prepare(){
        cout<<"Students won't prepare seriously for mid"<<endl;
    }
};
int main(){
    mid k;
    k.prepare();
    Sem a;
    a.prepare();
}